Neos.ContentRepository.TestSuite/Classes/Behavior/Features/Bootstrap/CRTestSuiteRuntimeVariables.php

```diff
@@ -160,8 +161,13 @@ public function visibilityConstraintsAreSetTo(string $restrictionType): void
 
     public function getCurrentSubgraph(): ContentSubgraphInterface
     {
-        return $this->currentContentRepository->getContentGraph()->getSubgraph(
-            $this->currentContentStreamId,
+        $contentGraphFinder = $this->currentContentRepository->projectionState(ContentGraphFinder::class);
+        $contentGraphFinder->reset();
+        if (isset($this->currentContentStreamId)) {
+            return $contentGraphFinder->getByWorkspaceNameAndContentStreamId($this->currentWorkspaceName, $this->currentContentStreamId)->getSubgraph($this->currentDimensionSpacePoint, $this->currentVisibilityConstraints);
+        }
+
+        return $contentGraphFinder->getByWorkspaceName($this->currentWorkspaceName)->getSubgraph(
             $this->currentDimensionSpacePoint,
             $this->currentVisibilityConstraints
```

Comment:
that seems kindof crazy and too intelligent ^^ .. is the `currentContentStreamId` set in the majority of the test or if not all? Then the actual code below is just dummy.

Reply:
There are tests that only refer to be in the workspace "XYZ" and it's current content stream, in that case only the workspace could be set.

Reply:
I see there are 5 different step syntaxes to change the workspace and or content stream id.

| Step | Modifies | Usages |
|--------|--------|--------|
| `I am in workspace "live"` | `currentWorkspaceName` | 8 |
| `I am in content stream "cs-id"` or <br> `I am in content stream "cs-id" and dimension space point {}`  | `currentContentStreamId` | 113 |
| `I am in the active content stream of workspace "live"` or <br> `I am in the active content stream of workspace "live" and dimension space point {}`  | `currentWorkspaceName` & `currentContentStreamId` | 788 | 

We hardly use the logic you introduced - only 8 times - as otherwise the `currentContentStreamId` will be set.

Also problematic, as you pointed out in [slack](https://neos-project.slack.com/archives/C04PYL8H3/p1714846896689819), is that the step `I am in content stream "cs-id" ...` will no longer be supported really as api ...

I dont know how what the plan is cc @nezaniel

Reply:
If we don’t communicate content stream ids, we don’t need them in the test runtime variables either

Reply:
Right, but that is a change for another day I guess

Reply:
Absolutely, yes.
Recommended way forward:
1) merge this
2) remove the remainig usages of currentContentSteamId in PHP, if any
3) rewrite all the test "given"s